test/tests/functional/pbs_sched_runjobwait.py

```diff
@@ -300,13 +302,13 @@ def test_throughput_ok(self):
 
         self.server.manager(MGR_CMD_SET, SCHED, {'throughput_mode': "False"},
                             id="default")
+        t1 = time.time()
```

Comment:
you shouldn't have had to move this, scheduling is off so submitting the job wouldn't run a cycle. Was that not the case?

Reply:
You are right Ravi, after you pointed, I checked it in detail and below are the testing results. I suspect test framework bug here; since the placing of time stamp code after job submission was right. I will log a framework ticket accordingly.

**Manual testing: Throughput_mode=false======================================**

**Server logs post job submission:**

03/25/2021 04:38:07.006940;0100;Server@sgkvm1;Req;;Type 0 request received from saritah@sgkvm1, sock=21
03/25/2021 04:38:07.009175;0100;Server@sgkvm1;Req;;Type 95 request received from saritah@sgkvm1, sock=23
03/25/2021 04:38:07.009472;0100;Server@sgkvm1;Req;;Type 21 request received from saritah@sgkvm1, sock=21
03/25/2021 04:38:07.009701;0100;Server@sgkvm1;Req;;Type 1 request received from saritah@sgkvm1, sock=21
03/25/2021 04:38:07.010281;0100;Server@sgkvm1;Job;45.sgkvm1;enqueuing into workq, state Q hop 1
03/25/2021 04:38:07.010981;0008;Server@sgkvm1;Job;45.sgkvm1;Job Queued at request of saritah@sgkvm1, owner = saritah@sgkvm1, job name = STDIN, queue = workq

**Server logs after running scheduling cycle:**

03/25/2021 04:38:40.685420;0100;Server@sgkvm1;Req;;Type 0 request received from root@sgkvm1, sock=23
03/25/2021 04:38:40.688599;0100;Server@sgkvm1;Req;;Type 95 request received from root@sgkvm1, sock=24
03/25/2021 04:38:40.688985;0100;Server@sgkvm1;Req;;Type 9 request received from root@sgkvm1, sock=23
03/25/2021 04:38:40.690224;0004;Server@sgkvm1;Svr;Server@sgkvm1;attributes set:  at request of root@sgkvm1
03/25/2021 04:38:40.690241;0004;Server@sgkvm1;Svr;Server@sgkvm1;attributes set: scheduling = t
03/25/2021 04:38:40.690286;0040;Server@sgkvm1;Svr;sgkvm1;Scheduler sent command 5
03/25/2021 04:38:40.690488;0100;Server@sgkvm1;Req;;Type 82 request received from Scheduler@sgkvm1, sock=20
03/25/2021 04:38:40.691114;0100;Server@sgkvm1;Req;;Type 21 request received from Scheduler@sgkvm1, sock=20
03/25/2021 04:38:40.691282;0100;Server@sgkvm1;Req;;Type 71 request received from Scheduler@sgkvm1, sock=20
03/25/2021 04:38:40.691403;0100;Server@sgkvm1;Req;;Type 58 request received from Scheduler@sgkvm1, sock=20
03/25/2021 04:38:40.691628;0100;Server@sgkvm1;Req;;Type 20 request received from Scheduler@sgkvm1, sock=20
**03/25/2021 04:38:40.691782;0100;Server@sgkvm1;Req;;Type 51 request received from Scheduler@sgkvm1, sock=20**
03/25/2021 04:38:40.692009;0100;Server@sgkvm1;Req;;Type 15 request received from Scheduler@sgkvm1, sock=20
03/25/2021 04:38:40.692035;0100;Server@sgkvm1;Node;sgkvm1;set_vnode_state;vnode.state=0x10 vnode_o.state=0x0 vnode.last_state_change_time=1616672320 vnode_o.last_state_change_time=1616671903 state_bits=0x10 state_bit_op_type_str=Nd_State_Or state_bit_op_type_enum=1
03/25/2021 04:38:40.692077;0008;Server@sgkvm1;Job;45.sgkvm1;Job Run at request of Scheduler@sgkvm1 on exec_vnode (sgkvm1:ncpus=1)

**PTL testing:==================================================** 

**Failed PTL test logs:**

2021-03-25 04:47:09,363 INFOCLI  sgkvm1(run_cmd): /opt/pbs/bin/qstat -f 47.sgkvm1
2021-03-25 04:47:09,374 INFO     expect on server sgkvm1: job_state = R && substate = 42 job 47.sgkvm1 attempt: 4 ...  OK
2021-03-25 04:47:09,379 INFO     server sgkvm1 log match: searching for "Type 15 request received" - with existence - **from 2021/03/25 04:47:07 - No match**
2021-03-25 04:47:11,313 INFO     server sgkvm1 log match: searching for "Type 15 request received" - with existence - from 2021/03/25 04:47:07 - attempt 2
2021-03-25 04:47:13,320 INFO     server sgkvm1 log match: searching for "Type 15 request received" - with existence - from 2021/03/25 04:47:07 - attempt 3

**Server log messages:**

03/25/2021 04:47:07.109832;0100;Server@sgkvm1;Req;;Type 51 request received from Scheduler@sgkvm1, sock=20
**03/25/2021 _04:47:07.110036_;0100;Server@sgkvm1;Req;;Type 15 request received from Scheduler@sgkvm1, sock=20**
03/25/2021 04:47:07.110057;0100;Server@sgkvm1;Node;sgkvm1;set_vnode_state;vnode.state=0x10 vnode_o.state=0x0 vnode.last_state_change_time=1616672827 vnode_o.last_state_change_time=1616672826 state_bits=0x10 state_bit_op_type_str=Nd_State_Or state_bit_op_type_enum=1
03/25/2021 04:47:07.110087;0008;Server@sgkvm1;Job;47.sgkvm1;Job Run at request of Scheduler@sgkvm1 on exec_vnode (sgkvm1:ncpus=1)
03/25/2021 04:47:07.123768;0100;Server@sgkvm1;Req;;Type 0 request received from saritah@sgkvm1, sock=23
03/25/2021 04:47:07.126222;0100;Server@sgkvm1;Req;;Type 95 request received from saritah@sgkvm1, sock=24

Reply:
ya, it's checking for the message from timestamp "2021/03/25 04:47:07", so it should have found the message with timestamp "03/25/2021 04:47:07.110036". Thanks for filing a framework ticket.